core/src/main/java/kafka/server/RemoteClusterMetadataManager.java

```diff
@@ -293,6 +288,30 @@ public void refreshRemoteMetadata() {
         updateConsumerGroupOffsets();
     }
 
+    private void maybeDeleteTopic(String clusterLinkName, Collection<MetadataResponse.TopicMetadata> topicMetadataResp) {
+        log.info("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!");
+        log.info("!!! periodic topicMetadataResp: {}", topicMetadataResp);
+        List<String> deletedTopics = new ArrayList<>();
+        // deleted topics if needed
+        topics.get(clusterLinkName).forEach(name -> {
+            List<String> remoteTopicNamesDeleted = topicMetadataResp.stream()
+                    .filter(topicMetadata -> topicMetadata.error() == Errors.UNKNOWN_TOPIC_OR_PARTITION)
+                    .map(MetadataResponse.TopicMetadata::topic).toList();
+
+            if (!topics.get(clusterLinkName).contains(name) || remoteTopicNamesDeleted.contains(name)) {
```

Comment:
I don't get this check `!topics.get(clusterLinkName).contains(name)` here. We are looping through the topics in clusterLinkName (i.e. in L296), and here, we want to check if this topic name is in `clusterLinkName`, isn't this check never true? Maybe we only keep `if (remoteTopicNamesDeleted.contains(name))` here?